### DISCUSSION TITLE
Error with "matching_columns"

### DISCUSSION THREAD

**Discussion by KidneyNode** [cafferychen777/ggpicrust2#41] [Q&A] [answered]
Having a problem with the following error, no matter how I format my metadata

Starting the ggpicrust2 analysis...
Converting KO to KEGG...
Calculation may take a long time, please be patient.
The kegg pathway with zero abundance in all the different samples has been removed.
Performing pathway differential abundance analysis...
Error in `metadata[, matching_columns]`:
! Can't subset columns with `matching_columns`.
✖ Subscript `matching_columns` can't contain missing values.
✖ It has a missing value at location 1.

**Comment by cafferychen777** [comment]
Hello @KidneyNode,

Based on the error message you provided, it seems that there is an issue with subsetting columns in the metadata tibble. The error specifically states that you cannot subset columns using the "matching_columns" argument, which should not contain any missing values.

To resolve this issue, please ensure the following:

1. Make sure that the column name for sample names in the metadata tibble is consistent with the column name in the abundance data.

For example, if the column name for sample names in the metadata is "SampleName", ensure that the abundance data also has a column with the same name ("SampleName") for sample names.

2. Check if there are any missing values in the "matching_columns" argument. The error message indicates that there is a missing value at location 1. Make sure that all values in the "matching_columns" argument are present and valid.

By addressing these points, you should be able to resolve the error related to subsetting columns in the metadata tibble.

If you have any further questions or encounter any additional issues, feel free to ask for assistance.

**Comment by KidneyNode** [comment]
Hi @cafferychen777, thanks so much for your response.

My metadata looks like this:
tibble("sample_name", "study_id", "timepoint") - with all sample_name entires unique

My abundance data is loaded from "R input/KO_pred_metagenome_unstrat.tsv", straight from the picrust2 output from python.
It does not have a column listed with sample names, but rather the sample names are the column names of the dataframe, with the rows as KO pathways. Does this need to be transposed?

I'm unsure what the "matching_columns" argument is, is is the sample_name column? There is no missing values in the dataframe/tibble

**Reply by KidneyNode** [reply, marked as answer]
OK, looks like I sorted it out by reading through your code for the pathway_daa function. The issue was that the metadata and abundance tibbles were of different lengths. Somehow my picrust2 output contained samples that I thought I had previously removed. By removing the extra samples from the abundance tibble, it then matched with the metadata, and the issue was resolved.
